src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -57,29 +57,26 @@ public static async Task<TypedData> ToRpc(this object value, ILogger logger, Grp
                 return typedData;
             }
 
-            if (value is byte[] arr)
-            {
-                typedData.Bytes = ByteString.CopyFrom(arr);
-            }
-            else if (value is JObject jobj)
-            {
-                typedData.Json = jobj.ToString(Formatting.None);
-            }
-            else if (value is string str)
-            {
-                typedData.String = str;
-            }
-            else if (value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities))
-            {
-                typedData = value.ToRpcCollection();
-            }
-            else if (value is HttpRequest request)
-            {
-                typedData = await request.ToRpcHttp(logger, capabilities);
-            }
-            else
-            {
-                typedData = value.ToRpcDefault();
+            switch (value)
+            {
+                case byte[] arr:
+                    typedData.Bytes = ByteString.CopyFrom(arr);
+                    break;
+                case JObject jobj:
+                    typedData.Json = jobj.ToString(Formatting.None);
+                    break;
+                case string str:
+                    typedData.String = str;
+                    break;
+                case double dbl:
+                    typedData.Double = dbl;
+                    break;
+                case HttpRequest request:
+                    typedData = await request.ToRpcHttp(logger, capabilities);
+                    break;
+                default:
+                    typedData = value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities) ? value.ToRpcCollection() : value.ToRpcDefault();
+                    break;
```

Comment:
Adding to the below, since this is an optimization. I noticed we're building and then discarding the initially created the `typedData` up top in the `HttpRequest` and default cases (existing issue) - we could avoid this allocation by moving the creations inline:
```cs
public static async Task<TypedData> ToRpc(this object value, ILogger logger, GrpcCapabilities capabilities)
{
    return value switch
    {
        null => new TypedData(),
        byte[] arr => new TypedData() { Bytes = ByteString.CopyFrom(arr) },
        JObject jobj => new TypedData() { Json = jobj.ToString(Formatting.None) },
        string str => new TypedData() { String = str },
        double dbl => new TypedData() { Double = dbl },
        HttpRequest request => await request.ToRpcHttp(logger, capabilities),
        _ => value.GetType().IsArray && IsTypedDataCollectionSupported(capabilities)
            ? value.ToRpcCollection()
            : value.ToRpcDefault(),
    };
}
```

Reply:
Switched to using inline creations for TypedData!

Reply:
Nice! I think we can simplify and improve even more here if you're game - just looked at the new version and noticed this is really a branched switch overall. We can collapse down and simplify if there's no reason not to - for example:

```cs
            switch (value)
            {
                case null:
                    return new TypedData();
                case byte[] arr:
                    return new TypedData() { Bytes = ByteString.CopyFrom(arr) };
                case JObject jobj:
                    return new TypedData() { Json = jobj.ToString(Formatting.None) };
                case string str:
                    return new TypedData() { String = str };
                case double dbl:
                    return new TypedData() { Double = dbl };
                case HttpRequest request:
                    return await request.ToRpcHttp(logger, capabilities);
                case byte[][] arrBytes when IsTypedDataCollectionSupported(capabilities):
                    return arrBytes.ToRpcByteArray();
                case string[] arrStr when IsTypedDataCollectionSupported(capabilities):
                    return arrStr.ToRpcStringArray();
                case double[] arrDouble when IsTypedDataCollectionSupported(capabilities):
                    return arrDouble.ToRpcDoubleArray();
                case long[] arrLong when IsTypedDataCollectionSupported(capabilities):
                    return arrLong.ToRpcLongArray();
                default:
                    return value.ToRpcDefault();
            }
```
...and eliminate the `ToRpcCollection` method doing resolution of type again. Thoughts?

In later versions of C#, that goes all the way down to:
```cs
        public static async Task<TypedData> ToRpc(this object value, ILogger logger, GrpcCapabilities capabilities) => value switch
        {
            null => new TypedData(),
            byte[] arr => new TypedData() { Bytes = ByteString.CopyFrom(arr) },
            JObject jobj => new TypedData() { Json = jobj.ToString(Formatting.None) },
            string str => new TypedData() { String = str },
            double dbl => new TypedData() { Double = dbl },
            HttpRequest request => await request.ToRpcHttp(logger, capabilities),
            byte[][] arrBytes when IsTypedDataCollectionSupported(capabilities) => arrBytes.ToRpcByteArray(),
            string[] arrStr when IsTypedDataCollectionSupported(capabilities) => arrStr.ToRpcStringArray(),
            double[] arrDouble when IsTypedDataCollectionSupported(capabilities) => arrDouble.ToRpcDoubleArray(),
            long[] arrLong when IsTypedDataCollectionSupported(capabilities) => arrLong.ToRpcLongArray(),
            _ => value.ToRpcDefault(),
        };
```
...if people find it readable, up to the team there on these kinds of things, wouldn't want to push them if not interested.

Overall, I think we may want to factor out `HttpRequest` specifically into another method if this is a hot path, because in those cases we're adding `async/await` overhead to a method where only one variant does so - we can potentially save additive overhead there (not proposing such in this PR, just noting something to look at).

Reply:
Makes sense to me, will update :D I did not even notice those two were related (not my PR originally, just helping someone else's code get merged 🙈🤦🏻‍♀️)

RE: HttpRequest, not sure if it is a hot path or not, but could be worth creating an optimization issue to look into it :)

Reply:
👍 To clarify: if `ToRpc()` at all is a hot path (without looking I'm guess "probably") - we're paying that overhead everywhere when we _could_ likely reduce is to only in the `HttpRequest` case - happy to poke when I start looking at out-of-proc if no one else has time first :)